### PR TITLE
P4-3405 updating dependenies. Changes are extensive due to breaking changes.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
This is a big PR due to breaking changes as a result of pulling in latest 3rd party dependencies.  A lot of the tests had to be updated due to packaging changes with Mockito.

The way commands are run has also had to change due to the CLI no longer working.  To be fair this does not come as a big surprise as it not really intended to be packaged withing a web application.  It would be packaged in its own application.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -57,15 +47,20 @@ dependencies {
...
     useJUnitPlatform()
     exclude("**/integration/*")
   }
 
-  withType<JavaCompile> {
-    sourceCompatibility = JavaVersion.VERSION_11.toString()
-    targetCompatibility = JavaVersion.VERSION_11.toString()
+  withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+    kotlinOptions {
+      jvmTarget = "17"
```
Comment (matt-crowson-moj) [inline build.gradle.kts:62]: Now that we have the latest HMPPS spring boot plugin and Kotlin plugin we can finally replace Java 11 with 17.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/cli/CommandRunner.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import uk.gov.justice.digital.hmpps.pecs.jpc.domain.price.Supplier
+import java.time.LocalDate
+
+/**
+ * This enables commands to be executed within the context of a running application (but not in WEB mode). Once a
+ * command has completed the application will terminate.
+ */
+@ConditionalOnNotWebApplication
+@Component
+class CommandRunner(
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/cli/CommandRunner.kt:17]: This replaces the Spring CLI.

File: build.gradle.kts [hunk 3]
```diff
@@ -15,39 +15,29 @@ dependencyCheck {
   suppressionFiles.add("calculate-journey-variable-payments-suppressions.xml")
 }
 
-// added specifically due to thymeleaf@3.0.12.RELEASE and CVE-2021-43466 - remove when update included in spring-boot
-ext["thymeleaf.version"] = "3.0.13.RELEASE"
```
Comment (matt-crowson-moj) [inline build.gradle.kts:19]: Hooray not longer needed.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/cli/ReportImportCommand.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import uk.gov.justice.digital.hmpps.pecs.jpc.service.ImportService
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+/**
+ * Allows manual running the importing of reporting data from move data that has been uploaded to S3.
+ */
+@ConditionalOnNotWebApplication
+@Component
+class ReportImportCommand(private val importService: ImportService) {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/cli/ReportImportCommand.kt:15]: This is mainly a rename.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/cli/CommandRunnerImporterCommandsTest.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.ContextConfiguration
+import uk.gov.justice.digital.hmpps.pecs.jpc.TestConfig
+import java.time.LocalDate
+
+@SpringBootTest(args = ["--report-import", "--from=2022-01-13", "--to=2022-01-14"], webEnvironment = WebEnvironment.NONE)
+@ActiveProfiles("test")
+@ContextConfiguration(classes = [TestConfig::class])
+class CommandRunnerImporterCommandsTest {
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/cli/CommandRunnerImporterCommandsTest.kt:17]: Two separate tests are needed for testing the commands when running as if the application was genuinely running due to the way the arguments for the test (i.e. identifying the command to run) are wired in at the class declaration, you cannot override at the individual test level.

**Review by Thource — APPROVED** [review]

**Pull request merged**